src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/AesImplementation.Browser.cs

```diff
@@ -2,12 +2,16 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
-using Internal.Cryptography;
 
 namespace System.Security.Cryptography
 {
     internal sealed partial class AesImplementation
     {
+        internal const int BlockSizeBytes = 16; // 128 bits
+
+        // SubtleCrypto doesn't support AES-192. http://crbug.com/533699
+        internal static readonly KeySizes[] s_legalKeySizes = { new KeySizes(128, 256, 128) };
```

Comment:
On one hand, I get it. On the other hand, I would be in favor of keeping legal key size as-is and instead throwing `PlatformNotSupportedException` if someone tries to use AES-192 in browsers.

This feels more aligned with how other algorithms handle this situation (DSA 186-4, ECDsa curves). Albeit those examples are all for symmetric algorithms.

Reply:
As a new-comer to this code, it feels wrong to provide incorrect key sizes and results for "IsValidKeySize". When we are executing on Browser, I guess I thought it was better to straight-up be honest about which key sizes are valid. If someone had code that inspected these properties/methods and returned a nice error to the user supplying the wrong key size, but now gets "PlatformNotSupportedException" doesn't seem like a great experience.

thoughts @bartonjs @GrabYourPitchforks ?

Reply:
The KeySizes value is supposed to represent the specific implementation/instance you're working with, not the algorithm abstractly.

RSACng reports that KeySizes have a step size of 64, because that's what CNG's documentation says: https://github.com/dotnet/runtime/blob/cbed235943395746786e4854f558a50d3b29033b/src/libraries/Common/src/System/Security/Cryptography/RSACng.cs#L35-L46

RSAOpenSsl reports it as a step size of 8: https://github.com/dotnet/runtime/blob/cbed235943395746786e4854f558a50d3b29033b/src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs#L70-L81

RSACryptoServiceProvider is from the stone ages and thinks RSA-384 is fine:  https://github.com/dotnet/runtime/blob/cbed235943395746786e4854f558a50d3b29033b/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RSACryptoServiceProvider.Windows.cs#L203-L209

Except on non-Windows, where it relays what the actual implementation has as limits: https://github.com/dotnet/runtime/blob/cbed235943395746786e4854f558a50d3b29033b/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs#L167-L169

So, not reporting AES-192 as valid via the LegalKeySizes seems correct to me.

Reply:
Sounds good then.